packages/eslint-plugin/lib/rules/no-use-is-focused.ts

```diff
@@ -0,0 +1,86 @@
+import type { Rule } from "eslint";
+import type { ImportDeclaration, CallExpression, Property } from "estree";
+
+// Custom Rule: No Import of useIsFocused from @react-navigation/native
+export const noUseIsFocusedImportRule: Rule.RuleModule = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Disallow importing `useIsFocused` from `@react-navigation/native` to encourage using `useFocusEffect` instead.",
+      category: "Best Practices",
+      recommended: true,
+      url: "https://github.com/bamlab/react-native-project-config/tree/main/packages/eslint-plugin/docs/rules/no-use-is-focused.md",
+    },
+    messages: {
+      noUseIsFocusedImport:
+        "Please use 'useFocusEffect' instead of 'useIsFocused' to avoid excessive rerenders.",
```

Comment:
```suggestion
        "Please use 'useFocusEffect' instead of 'useIsFocused' to avoid excessive rerenders: 'useIsFocused' will trigger rerender both when the page goes in and out of focus",
```